lxd/instance/drivers/driver_qemu.go

```diff
@@ -2902,13 +2909,14 @@ func (d *qemu) addCPUMemoryConfig(cfg *[]cfgSection) (int, error) {
 
 	cpuPinning := false
 
-	cpuCount, err := strconv.Atoi(cpus)
+	_, err := strconv.Atoi(cpus)
 	hostNodes := []uint64{}
 	if err == nil {
 		// If not pinning, default to exposing cores.
-		cpuOpts.cpuCount = cpuCount
+		// Only one CPU will be added here, as the others will be hotplugged during start.
+		cpuOpts.cpuCount = 1
```

Comment:
@monstermunchkin is there a reason why we cant add all of the CPUs via d.setCPUs() above, rather than doing the first one via a template and the rest via hotplugging?

Reply:
You mean that the qemu process wont start, even before virtualisation has been started?

Reply:
I don't recall exactly what didn't start, but one fixed CPU made it work.

Reply:
Hrm, can you try and see if we can add the 1st CPU via hotplugging after the QEMU process has started but before the VM has started.

The QEMU process is started here: https://github.com/lxc/lxd/blob/master/lxd/instance/drivers/driver_qemu.go#L1439
And the VM is started here: https://github.com/lxc/lxd/blob/master/lxd/instance/drivers/driver_qemu.go#L1546

Reply:
The NICs for instance are added as part of the pre-start mon hooks:

https://github.com/lxc/lxd/blob/master/lxd/instance/drivers/driver_qemu.go#L1518-L1525

Reply:
So, I tried it, and it does actually start. However, if we do

```
[smp-opts]
cpu = "0"
maxcpus = "8"
```

or omit `cpu`, QEMU will reserve `maxcpus`. That means, hotplugging won't work as those CPUs are fixed.

Reply:
So cpu = 0 is the same as cpu = 8 in this case?

Reply:
> So cpu = 0 is the same as cpu = 8 in this case?

Yes.